tests/kernel/fifo/fifo_timeout/src/main.c

```diff
@@ -152,9 +161,31 @@ static int test_multiple_threads_pending(struct timeout_order_data *test_data,
 			TC_PRINT(" thread (q order: %d, t/o: %d, fifo %p)\n",
 				data->q_order, data->timeout, data->fifo);
 		} else {
-			TC_ERROR(" *** thread %d woke up, expected %d\n",
-						data->timeout_order, ii);
-			return TC_FAIL;
+			/* Get the index of the thread which should have
+			 * actually timed out.
+			 */
+			for (j = 0; j < test_data_size; j++) {
+				if (test_data[j].timeout_order == ii) {
+					break;
+				}
+			}
+
+			if (data->timeout > test_data[j].timeout) {
```

Comment:
IMHO this check is not the one we want.
Whit this code, kernel might wake-up thread earlier than expected and the test will not notice that.

Reply:
That is already taken care in "is_timeout_in_range()" function. That function checks for the lower limit while the new code above checks for the upper limit.

Reply:
Ok